relayer/chains/processor/chain_processor.go

```diff
@@ -0,0 +1,24 @@
+package processor
+
+import (
+	"context"
+)
+
+// ChainProcessor interface is reponsible for polling blocks and emitting ibc message events to the PathProcessors
+// it is also responsible for tracking open channels and not sending messages to the PathProcessors for closed channels
+type ChainProcessor interface {
```

Comment:
Godoc must use complete sentences, with punctuation, so that documentation renders correctly.